### PR TITLE
Stored XSS in - franciscop/drive-db

### PR DESCRIPTION
**Pull request by b1nslashsh**
# Description

`franciscop/drive-db`. this package is vulnerable to `Stored Cross-Site Scripting (XSS)`.

https://github.com/franciscop/drive-db
https://www.npmjs.com/package/drive-db
# :recycle:  Steps To Reproduce-:  
  0) download https://github.com/franciscop/drive-db or use https://jsfiddle.net/v8uLpx59/1/
  1) Create A Google Drive spreadsheet and publish it:--->>(File > Publish to the Web > Publish Copy the id between /spreadsheets/ and /edit in the url)
  2) use this this for testing `https://docs.google.com/spreadsheets/d/13Sxv7P6yv6shDx_QfHnGoZ_Rt4ZSYO_x-9H_908vJis/edit?usp=sharing`
  3) then follow the steps in https://github.com/franciscop/drive-db to reproduce
# :male_detective: POC
poc

  [gdrive](https://drive.google.com/file/d/1VqfAgldmtY-qrgHRfizVvAH2oIFZVT-q/view?usp=sharing)
## 💥 Impact
run any javascript payloads

### CODE REVIEW

**Review by mzfr — COMMENTED** [review]
Just one thing:

```js
(async () => {
	const db = await drive({
  	sheet: '13Sxv7P6yv6shDx_QfHnGoZ_Rt4ZSYO_x-9H_908vJis',
    tab: '1'
  });
  console.log(db);
  document.body.innerHTML = `<pre>${JSON.stringify(db, null, 2)}</pre>`;
})();
```

This payload works and not the one which you refer to in your fiddle. That is because in the above code `tab:1` is set but in your fiddle code, the `tab:2` is set by default.

Using the code I posted above I was able to reproduce this.

Best,
mzfr

File: bounties/npm/package/drive-db/1/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        "PR": "N",
+        "UI": "R",
+        "S": "C",
+        "C": "L",
+        "I": "L",
+        "A": "N",
+        "E": "",
+        "RL": "",
+        "RC": "",
+        "Score": "8.1"
```
Comment (mzfr) [inline bounties/npm/package/drive-db/1/vulnerability.json:35]: I think severity should be lower than `8.1`

**Review by Mik317 — APPROVED** [review]
LGTM :smile: :cake:

Screenshot from 2020-12-16 15-56-16

@b1nslashsh please update the PoC code with the one provided by @mzfr :+1: 

PS: We should lower a bit since it's too high `8.1` (account takeover CVSS score mostly)

Cheers,
Mik

**Comment by b1nslashsh** [comment]
> LGTM :smile: :cake:
> 
> Screenshot from 2020-12-16 15-56-16
> 
> @b1nslashsh please update the PoC code with the one provided by @mzfr :+1: 
> 
> PS: We should lower a bit since it's too high `8.1` (account takeover CVSS score mostly)
> 
> Cheers,
> Mik

Hi @Mik317 , actually sorry for CVSS score am trying correct it on all my PRs,

**Comment by b1nslashsh** [comment]
> Just one thing:
> 
> ```js
> (async () => {
> 	const db = await drive({
>   	sheet: '13Sxv7P6yv6shDx_QfHnGoZ_Rt4ZSYO_x-9H_908vJis',
>     tab: '1'
>   });
>   console.log(db);
>   document.body.innerHTML = `<pre>${JSON.stringify(db, null, 2)}</pre>`;
> })();
> ```
> 
> This payload works and not the one which you refer to in your fiddle. That is because in the above code `tab:1` is set but in your fiddle code, the `tab:2` is set by default.
> 
> Using the code I posted above I was able to reproduce this.
> 
> Best,
> mzfr

Hey @mzfr ,
 Updated 👍

**Comment by Mik317** [comment]
> > LGTM smile cake
> > Screenshot from 2020-12-16 15-56-16
> > @b1nslashsh please update the PoC code with the one provided by @mzfr +1
> > PS: We should lower a bit since it's too high `8.1` (account takeover CVSS score mostly)
> > Cheers,
> > Mik
> 
> Hi @Mik317 , actually sorry for CVSS score am trying correct it on all my PRs,

No worries :smile:
Thanks for the quick response :+1: 

Cheers,
Mik

**Pull request closed without merging**